com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/XSStagingResolver.java

```diff
@@ -0,0 +1,64 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.CloudOperationException;
+import org.cloudfoundry.client.lib.StartingInfo;
+import org.cloudfoundry.client.lib.domain.PackageState;
+import org.springframework.http.HttpStatus;
+
+import com.sap.cloud.lm.sl.cf.process.Constants;
+import com.sap.cloud.lm.sl.cf.process.steps.ExecutionWrapper;
+import com.sap.cloud.lm.sl.cf.process.steps.StepsUtil;
+import com.sap.cloud.lm.sl.common.util.Pair;
+
+public class XSStagingResolver extends StagingResolver {
+
+    public XSStagingResolver(CloudControllerClient client) {
+        super(client);
+    }
+
+    @Override
+    public Pair<PackageState, String> getStagingState(ExecutionWrapper execution) {

```

Comment:
This is not a clean interface. getStagingState() returns the staging state, but it also logs what happened during the staging. Separate this in two methods: getStagingState() and getStagingLogs().